homeassistant/components/bluetooth/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Base config flow for a bluetooth integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.components import bluetooth
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+from .device import BluetoothDeviceData
+
+
+class BluetoothConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for a bluetooth device."""
+
+    VERSION = 1
+    DEVICE_DATA_CLASS = BluetoothDeviceData
+    USE_LOCAL_NAME = False
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovery_info: bluetooth.BluetoothServiceInfo | None = None
+        self._discovered_device: BluetoothDeviceData | None = None
+
+    async def async_step_bluetooth(
+        self, discovery_info: bluetooth.BluetoothServiceInfo
+    ) -> FlowResult:
```

Comment:
Maybe rename this and have it called from each integration so we can pass the `DEVICE_DATA_CLASS` and `use_local_name` there instead and make `async_step_bluetooth` an abstract method instead.

Then we can get rid of the mypy ignore